File: packages/backend/src/amini_server/schemas/registry.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AgentRegistryEntry(BaseModel):
    id: str
    agent_external_id: str
    name: str
    description: str | None = None
    framework: str = ""
    provider: str = ""
    risk_class: str = ""
    tags: list[str] | None = None
    data_access_patterns: dict | None = None
    deployment_status: str = "active"
    discovery_method: str = "sdk"
    regulations: list[str] | None = None
    session_count: int = 0
    violation_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AgentRegistryListResponse(BaseModel):
    agents: list[AgentRegistryEntry]
    total: int


class AgentRegistryUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    framework: str | None = None
    provider: str | None = None
    risk_class: str | None = None
    tags: list[str] | None = None
    data_access_patterns: dict | None = None
    deployment_status: str | None = None
    regulations: list[str] | None = None

File: packages/frontend/src/types/registry.ts
export interface AgentRegistryEntry {
  id: string
  agent_external_id: string
  name: string
  description: string | null
  framework: string
  provider: string
  risk_class: string
  tags: string[] | null
  data_access_patterns: Record<string, unknown> | null
  deployment_status: string
  discovery_method: string
  regulations: string[] | null
  session_count: number
  violation_count: number
  created_at: string
  updated_at: string
}

export interface AgentRegistryListResponse {
  agents: AgentRegistryEntry[]
  total: number
}
